core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessor.java

```diff
@@ -532,13 +537,13 @@ private static void sqlError(
     private void compileAndExecuteQuery(
             JsonQueryProcessorState state
     ) throws PeerDisconnectedException, PeerIsSlowToReadException, QueryPausedException, SqlException {
-        boolean recompileStale = true;
         try (SqlCompiler compiler = engine.getSqlCompiler()) {
-            for (int retries = 0; recompileStale; retries++) {
-                final long nanos = nanosecondClock.getTicks();
-                final CompiledQuery cc = compiler.compile(state.getQuery(), sqlExecutionContext);
+            for (int retries = 0; ; retries++) {
+                final long compilationStart = nanosecondClock.getTicks();
+                String query = state.getQuery().toString();
```

Comment:
why do we generate string here?

Reply:
I should remove all changes to `JsonQueryProcessor`. This is where the prototype code used to live, I obviousyl didn't do a great cleanup job.